js/app.js

```diff
@@ -48,10 +50,18 @@ $(document).ready(function() {
 
   // Input binding
   $searchInput
-  .on('keyup', function() {
-    var query = $(this).val();
-    toggleIconEmptyInput(query);
-    algoliaHelper.setQuery(query).search();
+  .on('keyup paste cut change', function() {
+    var $el = $(this);
+
+    // We use setTimeout to defer the callback to the bottom of the stack to
+    // catch the value after the paste, cut, or delete via mouse
+    setTimeout(function(){
+      var query = $el.val();
```

Comment:
If I udnerstand well, we have a callback to handle the "paste, cut or delete via mouse" because those are not events we can register to?

What about this: http://stackoverflow.com/a/8167009/147079

```js
$('#txt_field').bind('input propertychange', function() {
   console.log($(this).val());
});
```

If we can get out of the timer it would be cool because timers can be tricky and generate bugs also (like typing fast, all the timers will stackup, it can be tricky).